controllers/storageclassrequest/storageclassrequest_controller.go

```diff
@@ -131,6 +134,24 @@ func (r *StorageClassRequestReconciler) Reconcile(ctx context.Context, request r
 		return result, statusError
 	}
 
+	// wait for the cache to update so that we don't create multiple resources
```

Comment:
we have waitForsyncCache here https://github.com/red-hat-storage/ocs-operator/blob/13a9dfb60ebd8e539198a33e44ba22751e041c35/controllers/storageclassrequest/storageclassrequest_controller.go#L77-L79 isnt this helping?

Reply:
No, the create/update for status is done on the API server, the data is not written to the cache, the cache gets populated via a reflector that watches events from the API server. The apiserver might be generating the status update after we run over WaitForCacheSync.
We read the status in the second reconcile after WaitForCacheSync and it was empty leading to the creation of multiple svg

Reply:
@Madhu-1 waitForsyncCache is not doing what you expect it to do by looking at the name alone